### PR TITLE
Remove groovy scripting language

### PR DESCRIPTION
**Pull request by rjernst**
Groovy was deprecated in 5.0. This change removes it, along with the
legacy default language infrastructure in scripting.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/update.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
         },
         "lang": {
           "type": "string",
-          "description": "The script language (default: groovy)"
+          "description": "The script language (default: painless)"
```
Comment (jpountz) [inline rest-api-spec/src/main/resources/rest-api-spec/api/update.json]: maybe this part should be backported to 5.x

**Review by jpountz — APPROVED** [review]
It looks good to me. As a follow-up, maybe we should add a note to the 5.x percolator docs to tell users that percolator queries should use painless or expressions, or the index storing those queries won't load in 6.0.

**Comment by jdconrad** [comment]
@rjernst  This can not be checked in fast enough, are you busy or something :)?

File: rest-api-spec/src/main/resources/rest-api-spec/api/update.json (hunk 1 above)
Comment (rjernst) [inline rest-api-spec/src/main/resources/rest-api-spec/api/update.json]: I pushed 60f70bc to 5.x

**Pull request merged**